test_login.py

```diff
@@ -164,6 +165,15 @@ def test_login_disabled_is_set(self):
         login_manager = LoginManager(self.app, add_context_processor=True)
         self.assertFalse(login_manager._login_disabled)
 
+    def test_no_user_loader_raises(self):
+        login_manager = LoginManager(self.app, add_context_processor=True)
+        with self.app.test_request_context():
+            session['user_id'] = '2'
+            with self.assertRaises(Exception) as cm:
+                login_manager.reload_user()
+            expected_exception_message = 'No user_loader has been installed'
+            self.assertIn(expected_exception_message, str(cm.exception))
```

Comment:
This assertion feels a little loose, but that's not a huge deal. Maybe something like `self.assertIs(str(exception).startswith(expected), True)` is a little more sound.